### PR TITLE
feature(core/lifecycle-event): emits `render` after all render

### PR DESCRIPTION
**Pull request by bobylito**
When all the widgets are finished rendering and for each search response.

### CODE REVIEW

**Comment by bobylito** [comment]
Ready to review ;)

**Comment by vvo** [comment]
:heart_eyes: LGTM

File: lib/__tests__/InstantSearch-test.js [hunk 1]
```diff
@@ -176,4 +176,56 @@ describe('InstantSearch lifecycle', () => {
       expect(algoliasearchHelper.args[0][2].values).toEqual([-2, -1, 0, 1, 3, 4, 2]);
     });
   });
+
+  context('when render happens', () => {
+    var renderCalled;
+    beforeEach(() => {
+      renderCalled = 0;
```
Comment (vvo) [inline lib/__tests__/InstantSearch-test.js]: What about using sinon.js for spies like the previous tests are using? Of course the current spy code
you did is really simple and clean but let's use the tools we have even if there's a learning curve?

```js
var renderCalled = sinon.spy()
// ...
return {
  render: renderCalled
}

expect(renderCalled.callCount).toEqual(0);
```

Same thing for renderEmitted.

Comment (vvo) [inline lib/__tests__/InstantSearch-test.js]: And you can also check that render event was called AFTER render call :D `expect(renderCalled.calledBefore(renderEmitter));`

Comment (bobylito) [inline lib/__tests__/InstantSearch-test.js]: That's a neat feature there :) Will do the changes.

Comment (bobylito) [inline lib/__tests__/InstantSearch-test.js]: Btw there is `spy.reset()` seems handy to call in `beforeEach`.

File: lib/__tests__/InstantSearch-test.js [hunk 2]
```diff
@@ -176,4 +176,54 @@ describe('InstantSearch lifecycle', () => {
       expect(algoliasearchHelper.args[0][2].values).toEqual([-2, -1, 0, 1, 3, 4, 2]);
     });
   });
+
+  context('when render happens', () => {
+    var render = sinon.spy();
+    beforeEach(() => {
+      render.reset();
```
Comment (vvo) [inline lib/__tests__/InstantSearch-test.js]: yes you can use reset but jyou could also assign render only in beforeEach which result in the same (but avoid initiliazing it and then reset it). That's the same effect so don't care

**Comment by bobylito** [comment]
@vvo good to go?

**Pull request merged**